### PR TITLE
sink(ticdc): optimize the code structure of sink module

### LINKED ISSUE #5138: Optimize the overall code structure of sink module

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #5138

### What is changed and how it works?
1. split sink/mysql and sink/mq pkg.
```bash
.
├── codec
│   ├── craft
│   ├── avro_test.go
│   ├── canal_flat_test.go
│   ├── canal_test.go
│   ├── codec_test.go
│   ├── config_test.go
│   ├── craft_test.go
│   ├── interface_test.go
│   ├── json_test.go
│   ├── maxwell_test.go
│   ├── schema_registry_test.go
│   ├── avro.go
│   ├── canal_flat.go
│   ├── canal.go
│   ├── config.go
│   ├── craft.go
│   ├── interface.go
│   ├── java.go
│   ├── json.go
│   ├── maxwell.go
│   └── schema_registry.go
├── flowcontrol
│   ├── flow_control.go
│   └── flow_control_test.go
├── metrics
│   ├── metrics.go
│   └── statistics.go
├── mq
│   ├── dispatcher
│   ├── manager
│   ├── mq_flush_worker.go
│   ├── mq_flush_worker_test.go
│   ├── mq.go
│   ├── mq_test.go
│   └── producer
├── mysql
│   ├── causality.go
│   ├── causality_test.go
│   ├── mysql.go
│   ├── mysql_params.go
│   ├── mysql_params_test.go
│   ├── mysql_syncpoint_store.go
│   ├── mysql_test.go
│   ├── mysql_worker.go
│   ├── mysql_worker_test.go
│   ├── simple_mysql_tester.go
│   ├── syncpointStore.go
│   ├── txn_cache.go
│   ├── txn_cache_test.go
│   ├── txns_heap.go
│   └── txns_heap_test.go
├── black_hole.go
├── buffer_sink.go
├── buffer_sink_test.go
├── manager.go
├── manager_test.go
├── sink.go
├── sink_test.go
└── table_sink.go
```
2. Speed ​​up `TestExecDMLRollbackErrRetryable` by using smaller MaxRetry

before:
image

after:
image

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Breaking backward compatibility

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by hi-rustin** [comment]
Great work! Thanks for working on this.

> ```shell
> ├── common
> │   ├── flow_control.go
> │   ├── flow_control_test.go
> │   ├── metrics.go
> │   └── statistics.go
> ```

I think we can delete the common and create a new pkg like flow control. the rest is fine in the root directory. common/util is not a good name in most cases and is subject to abuse.

**Comment by asddongmen** [comment]
```
├── manager.go
├── manager_test.go
```
Do you think `sink_manager` would be a better name? It seems that there are too many `manager` in our repository.

**Comment by CharlesCheung96** [comment]
> ```shell
> ├── metrics
> │   ├── metrics.go
> │   └── statistics.go
> ```

Good suggestion. However, we still need to keep `metrics` separate in a pkg to avoid `cycle import` error. Otherwise, we need to create a metric file in the `sink/mysql` package.

**Comment by CharlesCheung96** [comment]
> ```
> ├── manager.go
> ├── manager_test.go
> ```
> 
> Do you think `sink_manager` would be a better name? It seems that there are too many `manager` in our repository.

Yes, it looks good to me.

File: cdc/sink/mq/dispatcher/event_router_test.go [hunk 1]
```diff
@@ -16,10 +16,11 @@ package dispatcher
 import (
 	"testing"
 
+	partition2 "github.com/pingcap/tiflow/cdc/sink/mq/dispatcher/partition"
+	"github.com/pingcap/tiflow/cdc/sink/mq/dispatcher/topic"
+
```
Comment (hi-rustin) [inline cdc/sink/mq/dispatcher/event_router_test.go:21]:
```suggestion
	partition2 "github.com/pingcap/tiflow/cdc/sink/mq/dispatcher/partition"
	"github.com/pingcap/tiflow/cdc/sink/mq/dispatcher/topic"
```
ditto

File: cdc/sink/mq/manager/kafka/manager_test.go [hunk 2]
```diff
@@ -17,7 +17,8 @@ import (
 	"testing"
 	"time"
 
-	kafkaconfig "github.com/pingcap/tiflow/cdc/sink/producer/kafka"
+	kafkaconfig "github.com/pingcap/tiflow/cdc/sink/mq/producer/kafka"
+
```
Comment (hi-rustin) [inline cdc/sink/mq/manager/kafka/manager_test.go:21]:
```suggestion
	kafkaconfig "github.com/pingcap/tiflow/cdc/sink/mq/producer/kafka"
```

File: cdc/sink/mq/mq.go [hunk 3]
```diff
@@ -11,26 +11,28 @@
...
 
+	"github.com/pingcap/tiflow/cdc/sink/metrics"
+	"github.com/pingcap/tiflow/cdc/sink/mq/dispatcher"
+	"github.com/pingcap/tiflow/cdc/sink/mq/manager"
+	kafkamanager "github.com/pingcap/tiflow/cdc/sink/mq/manager/kafka"
+	pulsarmanager "github.com/pingcap/tiflow/cdc/sink/mq/manager/pulsar"
+	"github.com/pingcap/tiflow/cdc/sink/mq/producer"
+	kafka2 "github.com/pingcap/tiflow/cdc/sink/mq/producer/kafka"
+	"github.com/pingcap/tiflow/cdc/sink/mq/producer/pulsar"
+
```
Comment (hi-rustin) [inline cdc/sink/mq/mq.go:30]:
```suggestion
	"github.com/pingcap/tiflow/cdc/sink/metrics"
	"github.com/pingcap/tiflow/cdc/sink/mq/dispatcher"
	"github.com/pingcap/tiflow/cdc/sink/mq/manager"
	kafkamanager "github.com/pingcap/tiflow/cdc/sink/mq/manager/kafka"
	pulsarmanager "github.com/pingcap/tiflow/cdc/sink/mq/manager/pulsar"
	"github.com/pingcap/tiflow/cdc/sink/mq/producer"
	kafka2 "github.com/pingcap/tiflow/cdc/sink/mq/producer/kafka"
	"github.com/pingcap/tiflow/cdc/sink/mq/producer/pulsar"
```

File: cdc/sink/mq/mq_flush_worker.go [hunk 4]
```diff
@@ -11,17 +11,19 @@
...
-package sink
+package mq
 
 import (
 	"context"
 	"time"
 
+	"github.com/pingcap/tiflow/cdc/sink/metrics"
+	"github.com/pingcap/tiflow/cdc/sink/mq/producer"
+
```
Comment (hi-rustin) [inline cdc/sink/mq/mq_flush_worker.go:22]:
```suggestion
	"github.com/pingcap/tiflow/cdc/sink/metrics"
	"github.com/pingcap/tiflow/cdc/sink/mq/producer"
```

File: cdc/sink/sink.go [hunk 5]
```diff
@@ -18,19 +18,17 @@ import (
 	"net/url"
 	"strings"
 
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/tiflow/cdc/sink/mq"
+
```
Comment (hi-rustin) [inline cdc/sink/sink.go:23]:
```suggestion
	"github.com/pingcap/failpoint"
	"github.com/pingcap/tiflow/cdc/sink/mq"
```

File: cdc/sink/mq/dispatcher/event_router.go [hunk 6]
```diff
@@ -16,11 +16,12 @@ package dispatcher
 import (
 	"strings"
 
+	partition2 "github.com/pingcap/tiflow/cdc/sink/mq/dispatcher/partition"
+	topic2 "github.com/pingcap/tiflow/cdc/sink/mq/dispatcher/topic"
+
```
[the remark this answers is not in the document]
Comment (hi-rustin) [reply cdc/sink/mq/dispatcher/event_router.go:21]: Also partition2 not a good name.

[the remark this answers is not in the document]
Comment (CharlesCheung96) [reply cdc/sink/mq/dispatcher/event_router.go:21]: This is a mistake, we don't need an alias

File: cdc/sink/table_sink.go [hunk 7]
```diff
@@ -88,7 +88,12 @@ func (t *tableSink) flushResolvedTs(ctx context.Context, resolvedTs uint64) (uin
 	if redoTs < resolvedTs {
 		resolvedTs = redoTs
 	}
-	return t.manager.flushBackendSink(ctx, t.tableID, resolvedTs)
+
```
Comment (CharlesCheung96) [inline cdc/sink/table_sink.go:91]: There is a function adjustment involved here.

**Review by asddongmen — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

File: cdc/sink/table_sink.go (hunk 7 above)
Comment (hi-rustin) [reply cdc/sink/table_sink.go:91]: Is there any particular reason to modify it? Does it feel like the original is better encapsulated?

Comment (CharlesCheung96) [reply cdc/sink/table_sink.go:91]: There is no special reason, just to make it easier to read the code.

**Pull request merged**